Machines/PCCompatible/RTC.hpp

```diff
@@ -0,0 +1,96 @@
+//
+//  RTC.hpp
+//  Clock Signal
+//
+//  Created by Thomas Harte on 06/12/2023.
+//  Copyright © 2023 Thomas Harte. All rights reserved.
+//
+
+#ifndef RTC_h
+#define RTC_h
+
+#include <ctime>
+
+namespace PCCompatible {
+
+class RTC {
+	public:
+		template <int address>
+		void write(uint8_t value) {
+			switch(address) {
+				default: break;
+				case 0:
+					selected_ = value & 0x7f;
+					// NMI not yet supported.
+				break;
+				case 1:
+					write_register(value);
+				break;
+			}
+		}
+
+		uint8_t read() {
+			std::time_t now = std::time(NULL);
+			std::tm *time_date = std::localtime(&now);
+
+			switch(selected_) {
+				default:
+				return 0xff;
+
+				case 0x00:	return bcd(time_date->tm_sec);			// Seconds [0-59]
+				case 0x01:	return 0;	// Seconds alarm
+				case 0x02:	return bcd(time_date->tm_min);			// Minutes [0-59]
+				case 0x03:	return 0;	// Minutes alarm
+				case 0x04:
+					// Hours [1-12 or 0-23]
+					if(statusB_ & 2) {
+						return bcd(time_date->tm_hour);
+					}
+					return
+						// TODO: determine why GLaTICK signals an RTC fault if I enable communication of PM as below.
+//						((time_date->tm_hour >= 12) ? 0x80 : 0x00) |
+						bcd(1 + (time_date->tm_hour + 11)%12);
+				break;
+				case 0x05:	return 0;	// Hours alarm
+				case 0x06:	return bcd(time_date->tm_wday + 1);		// Day of the week [Sunday = 1]
+				case 0x07:	return bcd(time_date->tm_mon);			// Date of the Month [1-31]
```

Comment:
Returning `tm_mon` for Date of the Month? Why not return `tm_mday`?

Reply:
Ugh, you're right. My enthusiasm at having a little time here seems to have destroyed any sense of quality control. See also the nonsense of the CGA snow flag originally being tied to horizontal sync rather than display enable.

Anyway, fixed in https://github.com/TomHarte/CLK/pull/1266